server/views/script.js

```diff
@@ -0,0 +1,224 @@
+    rowLetter += `<div class="letterCell"><div id="${i}_letter" class="letter">${String.fromCharCode(charCodeAt)}</div></div>`;
+    charCodeAt += 1;
+  }
+  rowLetter += '</div>';
+  html += rowLetter;
+
+  html += '<form name="extable" class="extable">';
+  csv.forEach((row, rowIndex) => {
+    html += '<div class="row">';
+    html += `<div class="numberRow" id="${rowIndex}_number"><div class="number">${rowNumber}</div></div>`;
+    rowNumber += 1;
+    row.forEach((cell, cellIndex) => {
+      html += `<div class="cell"> <input id="${rowIndex}_${cellIndex}" onmouseover="backlight('${rowIndex}_${cellIndex}', '#cacaca')" onmouseout="backlight('${rowIndex}_${cellIndex}', '#f3f3f5')" class="cellInput cell" onclick="openInput(this)" onkeydown = "if (event.keyCode == 13) calculate()" type="text" name="cell" value="${cell}"> </div>`;
+    });
+    html += '</div>';
+  });
+  html += `<input type="hidden" name="rowCount" value="${csv.length}">`;
+  html += '</form>';
+  return html;
+}
+
+function printHtml(html) {
+  document.getElementById('table').innerHTML = html;
+}
+
+  function calculate(filename) {
+    let formData = new FormData(document.forms.extable);
+    if (filename) {
+      formData.append("filename", filename);
+    }
+    let xhr = new XMLHttpRequest();
+    xhr.open("POST", "/calculate");
+    xhr.send(formData);
+
+    xhr.onload = () => printHtml(displayPage(xhr.response));
+  }
+
+  let input = document.getElementsByClassName('cellInputForm')[0];
+  function openInput(cell) {
+  input.value = cell.value;
+  input.setAttribute('cellId', cell.id);
+  changeInput(cell);
+  changeCell(cell);
+}
+
+function changeInput(cell) {
+  cell.oninput = function() {
+    input.value = cell.value;
+    input.setAttribute('cellId', cell.id);
+    highlightTheMentionedCell(cell.value);
+  };
+}
+
+function highlightTheMentionedCell(value) {
+  let cells = document.getElementsByClassName('cell');
+  for (let i = 0; i < cells.length; i++) {
+    cells[i].classList.remove("highlightRed");
+  }
+  let links = value.match(/[A-Z]\d+/g);
+  if (!links) {
+    return;
+  }
+  for (let i = 0; i < links.length; i++) {
+    const letter = links[i].match(/[A-Z]/)[0];
+    const number = links[i].match(/(\d+)/)[0];
+    let id = `${number - 1}_${letter.charCodeAt(0) - 65}`;
+    let linkCell = document.getElementById(id);
+    linkCell.classList.add("highlightRed");
+  }
+}
+
+function changeCell(cell) {
+  input.oninput = function() {
+    cell.value = input.value;
+    highlightTheMentionedCell(input.value);
+  };
+}
+
+function writeCell() {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  cell.value = input.value;
+  calculate();
+  setTimeout(writeInput, 1000);
+}
+
+function writeInput(argument) {
+  let cellId = input.getAttribute('cellId');
+  let cell = document.getElementById(cellId);
+  input.value = cell.value;
+}
+
+function backlight(id, color) {
+  let numberId = id.match(/(\d+)_/)[1];
```

Comment:
remember difference between let and const?

Reply:
done